cli/src/main/java/com/devonfw/tools/ide/environment/AbstractEnvironmentVariables.java

```diff
@@ -149,9 +149,25 @@ public AbstractEnvironmentVariables extend(Path propertiesFilePath, EnvironmentV
     return new EnvironmentVariablesPropertiesFile(this, type, propertiesFilePath, this.context);
   }
 
+  private EnvironmentVariablesType incrementType(EnvironmentVariablesType type) {
+
+    EnvironmentVariables current = this;
+    while (current.getType() != type) {
+      current = current.getParent();
+      if (current == null) {
+        throw new IllegalStateException("Could not find type " + type + " in " + this);
```

Comment:
can this happen? If so it will break our variable resolution and therefore prevent the user from starting the IDE causing a full blocker issue.
If we cannot resolve a variable it is better to leave it unresolved and e.g. log a warning but not fail hard.

Reply:
Okey, I changed it.